Server/mods/deathmatch/logic/luadefs/CLuaTeamDefs.cpp

```diff
@@ -67,282 +66,83 @@ void CLuaTeamDefs::AddClass(lua_State* luaVM)
     lua_registerclass(luaVM, "Team", "Element");
 }
 
-int CLuaTeamDefs::SetPlayerTeam(lua_State* luaVM)
+CTeam* CLuaTeamDefs::CreateTeam(lua_State* lua, std::string name, std::uint8_t red, std::uint8_t green, std::uint8_t blue) noexcept
 {
-    CPlayer* pPlayer;
-    CTeam*   pTeam;
+    CLuaMain* vm = g_pGame->GetLuaManager()->GetVirtualMachine(lua);
 
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pPlayer);
-    argStream.ReadUserData(pTeam, NULL);
+    if (!vm)
+        return nullptr;
 
-    if (!argStream.HasErrors())
-    {
-        if (CStaticFunctionDefinitions::SetPlayerTeam(pPlayer, pTeam))
-        {
-            lua_pushboolean(luaVM, true);
-            return 1;
-        }
-    }
-    else
-        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
+    CResource* resource = vm->GetResource();
 
-    lua_pushboolean(luaVM, false);
-    return 1;
-}
+    if (!resource)
+        return nullptr;
 
-int CLuaTeamDefs::CreateTeam(lua_State* luaVM)
-{
-    SString       strName;
-    unsigned char ucRed;
-    unsigned char ucGreen;
-    unsigned char ucBlue;
+    CTeam* team = CStaticFunctionDefinitions::CreateTeam(resource, name.c_str(), red, green, blue);
+
+    if (!team)
+        return nullptr;
 
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadString(strName);
-    argStream.ReadNumber(ucRed, 235);
-    argStream.ReadNumber(ucGreen, 221);
-    argStream.ReadNumber(ucBlue, 178);
+    CElementGroup* group = resource->GetElementGroup();
 
-    if (!argStream.HasErrors())
-    {
-        CLuaMain* pLuaMain = g_pGame->GetLuaManager()->GetVirtualMachine(luaVM);
-        if (pLuaMain)
-        {
-            CResource* pResource = pLuaMain->GetResource();
-            if (pResource)
-            {
-                CTeam* pTeam = CStaticFunctionDefinitions::CreateTeam(pResource, strName, ucRed, ucGreen, ucBlue);
-                if (pTeam)
-                {
-                    CElementGroup* pGroup = pResource->GetElementGroup();
-                    if (pGroup)
-                    {
-                        pGroup->Add(pTeam);
-                    }
-                    lua_pushelement(luaVM, pTeam);
-                    return 1;
-                }
-            }
-        }
-    }
-    else
-        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
+    if (group)
+        group->Add(team);
 
-    lua_pushboolean(luaVM, false);
-    return 1;
+    return team;
 }
 
-int CLuaTeamDefs::GetPlayersInTeam(lua_State* luaVM)
+CTeam* CLuaTeamDefs::GetTeamFromName(const std::string name) noexcept
```

Comment:
This method isn't backward compatible. Because originally `getTeamFromName` returns `false` on failure. But after this PR it will return `nil`.